### PR TITLE
[Tests]Remove UnfinishedVMIPodSelector from tests/utils.go 

### PR DESCRIPTION
**Pull request by Barakmor1**
Replace all uses of UnfinishedVMIPodSelector in tests/utils.go with libpod to improve consistency in the code.

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/vmi_hook_sidecar_test.go [hunk 1]
```diff
@@ -189,16 +177,12 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
...
 				Expect(err).NotTo(HaveOccurred(), "the request to create the VMI should be accepted")
 
 				Eventually(func() bool {
-					vmiPod, exists, err := getVMIPod(vmi)
-					if err != nil {
-						Expect(err).NotTo(HaveOccurred(), "must be able to retrieve the VMI virt-launcher pod")
-					} else if !exists {
-						return false
-					}
+					vmiPod, err := libpod.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
```
Comment (ShellyKa13) [inline tests/vmi_hook_sidecar_test.go:184]: Just making sure, when the VM is scheduling does it mean the vmi pod has to exist already?

**Review by ShellyKa13 — COMMENTED** [review]
Nice!
One small question
/lgtm

File: tests/vmi_hook_sidecar_test.go (hunk 1 above)
Comment (enp0s3) [reply tests/vmi_hook_sidecar_test.go:184]: @ShellyKa13 Hi, yep, from code snippet:
```
	// A target Pod exists but is not yet scheduled and in running state.
	Scheduling VirtualMachineInstancePhase = "Scheduling"
```

**Review by enp0s3 — COMMENTED** [review]
/approve
@Barakmor1 Thanks!

**Review by dankenigsberg — APPROVED** [review]

**Pull request merged**